lib/utils.coffee

```diff
@@ -39,9 +40,16 @@ setTabIndex = (elems) ->
 # ==================================================
 # Template
 #
+dirTemplate = (directory, date, editor) ->
+  if editor
+    @frontMatter = new FrontMatter(editor)
+    title = dasherize(@frontMatter.get("title") || getTitleSlug(editor.getPath()))
+    category = @frontMatter.get("category")
+  info =
+    title: title || dasherize("New Post")
+    category: category || "uncategorized"
```

Comment:
I would prefer the caller pass in additional data that can be used in, it is easier to use in future as well.

```
dirTemplate = (directory, date, data = {}) ->

# in image view
siteImagesDir: -> utils.dirTemplate(config.get("siteImagesDir"), { slug: ... })
```